File: XCase.ProxyGenerator/REST/Operation.cs
namespace XCase.ProxyGenerator.REST
{
    using System.Collections.Generic;

    public class Operation
    {
        public Operation(string returnType, string method, string path, List<Parameter> parameters, string operationId, string description, string proxyName)
        {
            this.Path = path;
            this.Method = method;
            this.Parameters = parameters;
            this.OperationId = operationId;
            this.Description = description;
            this.ReturnType = returnType;
            this.ProxyName = proxyName;
        }

        public string ProxyName { get; set; }
        public string Path { get; set; }
        public string Method { get; set; }
        public List<Parameter> Parameters { get; set; }
        public string OperationId { get; set; }
        public string Description { get; set; }
        public string ReturnType { get; set; }

        public static string GetCleanJavaReturnType(string returnType)
        {
            bool isArray = false;
            if (!string.IsNullOrEmpty(returnType))
            {
                if (returnType.EndsWith("[]"))
                {
                    isArray = true;
                    returnType = GetCleanJavaReturnType(returnType.Substring(0, returnType.Length - 2));
                }

                if (returnType == "object")
                {
                    return "Object";
                }
                else if (returnType == "string")
                {
                    return "String";
                }

                if (isArray)
                {
                    return string.Format("{0}[]", returnType);
                }

                return returnType;
            }

            return returnType;
        }
    }
}

File: platform/src/java/com/xcase/rest/generator/Operation.java
package com.xcase.rest.generator;

import com.xcase.common.impl.simple.core.CommonHTTPManager;
import java.lang.invoke.MethodHandles;
import java.util.*;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class Operation
{
    public Operation(String returnType, String method, String path, List<Parameter> parameters, String operationId, String description, String proxyName) {
        this.Path = path;
        this.Method = method;
        this.Parameters = parameters;
        this.OperationId = operationId;
        this.Description = description;
        this.ReturnType = returnType;
        this.ProxyName = proxyName;
    }

    public String ProxyName;
    public String Path;
    public String Method;
    public List<Parameter> Parameters;
    public String OperationId;
    public String Description;
    public String ReturnType;

    public static String getCleanJavaReturnType(String returnType) {
        boolean isArray = false;
        if (returnType != null && !returnType.isEmpty()) {
            if (returnType.endsWith("[]")) {
                isArray = true;
                returnType = getCleanJavaReturnType(returnType.substring(0, returnType.length() - 2));
            }

            if (returnType.equals("object")) {
                return "Object";
            }
            else if (returnType.equals("string")) {
                return "String";
            }

            if (isArray) {
                return returnType + "[]";
            }

            return returnType;
        }

        return returnType;
    }
}